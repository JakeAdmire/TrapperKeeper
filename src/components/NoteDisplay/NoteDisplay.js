import React from 'react';
import { PropTypes } from 'prop-types';

import checked from '../../media/checked.svg';
import unchecked from '../../media/unchecked.svg';

const NoteDisplay = (props) => {
  const titleMax = 13;
  const listMax = 20;
  return (
    <div className="NoteDisplay">
      <h3>
        {
          props.title.length > titleMax ?
          props.title.substring(0, titleMax) + '...' :
          props.title
        }
      </h3>
      <ul>
      {
        props.list.map(item => (
          <li key={item.id} className={item.isComplete ? 'completed' : 'incompelete'}>
            <div className="checkBox"><img src={item.isComplete ? checked : unchecked} alt="Check Box" /></div>
            {
              item.text.length > listMax ?
              item.text.substring(0, listMax) + '...' :
              item.text
            }
          </li>
        ))
      }
      </ul>
    </div>
  )

}

export default NoteDisplay;
