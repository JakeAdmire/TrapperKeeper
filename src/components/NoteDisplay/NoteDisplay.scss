.NoteDisplay {
  display: block;
  pointer-events: none;

  h3 {
    color: #e0e0e0;
    margin: 0;
    text-align: left;
  }

  ul {
    text-align: left;
    padding: 0;
    margin: 0;

    li {
      color: #e0e0e0;
      list-style-type: none;
      display: flex;
      justify-content: flex-start;
      align-items: flex-start;
      line-height: 1.2;

      &:hover {

      }

      .checkBox {
        opacity: .1;
        img {
          max-width: 75%;
        }
      }

    }

    li.completed {

    }

    li.incompelete {

    }

  }


}
