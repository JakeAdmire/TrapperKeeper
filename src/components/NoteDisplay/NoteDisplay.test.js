import React from 'react';
import NoteDisplay from './NoteDisplay';
import { shallow } from 'enzyme';

let mockId = "ZN9zKmsle"
let list = [{ id: "BqkLJDU45", isComplete: false, text: "Item 2" }]
let mockTitle = "Another 2";

describe('NoteDisplay', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <NoteDisplay  id={mockId}
                    list={list}
                    title={mockTitle} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

})