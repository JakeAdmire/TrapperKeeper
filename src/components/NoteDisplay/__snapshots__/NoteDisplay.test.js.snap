// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteDisplay should match the snapshot 1`] = `
<div
  className="NoteDisplay"
>
  <h3>
    Another 2
  </h3>
  <ul>
    <li
      className="incompelete"
      key="BqkLJDU45"
    >
      Item 2
    </li>
  </ul>
</div>
`;
